extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -1603,15 +1603,50 @@ public Function<TopicPartition, KafkaRecordConsumer> visit(@NotNull final PerPar
      *        {@link TableDefinition}. See {@link StreamConsumerRegistrarProvider#single(SingleConsumerRegistrar)
      *        single} and {@link StreamConsumerRegistrarProvider#perPartition(PerPartitionConsumerRegistrar)
      *        per-partition}.
+     * @return the newly created KafkaIngester, the caller must call start on the returned object to begin processing
+     *         messages
```

Comment:
This is a breaking change to an existing public API; I'd suggest returning void and internally starting the consume (same as old behavior). I think your new `KafkaIngester consume(...PartitionToInitialOffsetFunction...)` method should suffice for any new callers who want more explicit control?

Reply:
Worth pointing out, this API call was just added last release, specifically for DnD use cases.